autogen/cluster.tf

```diff
@@ -219,14 +219,75 @@ resource "google_container_cluster" "primary" {
 /******************************************
   Create Container Cluster node pools
  *****************************************/
+locals {
+  force_node_pool_recreation_resources = [
+    "disk_size_gb",
+    "disk_type",
+    "accelerator_count",
+    "accelerator_type",
+    "local_ssd_count",
+    "machine_type",
+    "preemptible",
+    "service_account",
+  ]
+}
+
+resource "random_id" "name" {
+  count       = var.node_pools_create_before_destroy ? length(var.node_pools) : 0
+  byte_length = 2
+  prefix      = format("%s-", lookup(var.node_pools[count.index], "name"))
+  keepers = merge(
+    zipmap(
+      local.force_node_pool_recreation_resources,
+      [for keeper in local.force_node_pool_recreation_resources : lookup(var.node_pools[count.index], keeper, "")]
+    ),
+    {
+      labels_all = join(",", keys(var.node_pools_labels["all"]), values(var.node_pools_labels["all"]))
```

Comment:
I think we need to combine `labels_all` and `labels_node_pool` into a single keeper. Otherwise moving a label from `node_pool` to `all` could incorrectly force a recreation. (Similar for metadata, oauth, etc.)

Reply:
Makes sense. Will update

Reply:
updated.